tools/c7n_azure/c7n_azure/resources/key_vault.py

```diff
@@ -58,9 +68,33 @@ def __call__(self, i):
                         'certificates': policy.permissions.certificates
                     }
                 })
-            # Enhance access policies with display_name, object_type and principal_name
+            # Enhance access policies with displayName, aadType and principalName
             i['accessPolicies'] = self.enhance_policies(access_policies)
-        return super(WhiteListFilter, self).__call__(i)
+
+        # Ensure each policy is
+        #   - User is whitelisted
+        #   - Permissions don't exceed allowed permissions
+        for p in i['accessPolicies']:
+            if p[self.key] not in self.users:
+                if not self.compare_policies(p['permissions'], self.permissions):
+                    return False
+        return True
+
+    @staticmethod
+    def compare_policies(user_permissions, permissions):
```

Comment:
Since the full filter is difficult to test, I would at least add a test for this static method.

Reply:
Done. (Also changed function name)